assets/js/components/surveys/SurveyQuestionRatingChoice.js

```diff
@@ -25,18 +25,26 @@ import PropTypes from 'prop-types';
  * WordPress dependencies
  */
 import { __, sprintf } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Button from '../../components/Button';
+import IconSurveyUnhappy from '../../../svg/survey-unhappy.svg';
+import IconSurveyDissatisfied from '../../../svg/survey-dissatisfied.svg';
+import IconSurveyNeutral from '../../../svg/survey-neutral.svg';
+import IconSurveySatisfied from '../../../svg/survey-satisfied.svg';
+import IconSurveyDelighted from '../../../svg/survey-delighted.svg';
 
-const SurveyQuestionRatingChoice = ( { Icon, choice, answerQuestion } ) => {
-	const handleButtonClick = () => {
+const SurveyQuestionRatingChoice = ( { choice, answerQuestion } ) => {
+	const handleButtonClick = useCallback( () => {
 		if ( typeof answerQuestion === 'function' ) {
 			answerQuestion( choice );
 		}
-	};
+	}, [ answerQuestion, choice ] );
+
+	const Icon = SurveyQuestionRatingChoice.ordinalIconMap[ choice.answer_ordinal ];
```

Comment:
This will result in an error below if an answer is passed with an ordinal that does not map to an icon. This _should_ never happen, but we should account for this case just to be safe since this comes entirely from an external source.

If there is no `Icon` then there will be nothing to click and no point in rendering anything else here, so we should probably just return `null` in this case.

```suggestion
	const Icon = SurveyQuestionRatingChoice.ordinalIconMap[ choice.answer_ordinal ];

	if ( ! Icon ) {
		return null;
	}
```

Reply:
Updated @aaemnnosttv